src/main/java/broker/LogSegment.java

```diff
@@ -0,0 +1,77 @@
+package broker;
+
+import org.tinylog.Logger;
+
+import java.io.File;
+import java.io.IOException;
+
+public class LogSegment {
+    private final int partitionNumber;
+    private File logFile; // stores the records
+    private boolean isActive; // if isActive is true, the segment is mutable
+    private long segmentThresholdInBytes =  1_048_576; // log segment cannot exceed this size threshold (default: 1 MB)
+    private long currentSizeInBytes;
+    private int startOffset; // for entire segment
```

Comment:
I see that the `segment threshold` var is a long but the `offset` vars are ints. Was just curious about that difference

Reply:
Yep I have the `offset` vars as ints because the real kafka defines the offsets as strictly increasing integers to uniquely ID each record, so theyre not referencing byte offsets which confused me at first. Thats why the threshold var is a long even though realistically for this system we'll probably never get past the largest int threshold

Reply:
Got it!